### ISSUE TITLE
pio_put/get_var fails for NULL start/count

### ISSUE THREAD

**Issue by jayeshkrishna** [NCAR/ParallelIO#204]
pio_put/get_var fails with a NULL start pointer, with the following error message,

Abort with message argument start is a NULL pointer in file /homes/jayesh/acme/ACME-test-pio2/cime/externals/pio2/src/clib/pio_getput_int.c at line 694

**Comment by jayeshkrishna** [comment]
I have a branch with a test for the case and am working on a fix.

**Comment by jedwards4b** [comment]
Why would the start pointer be null?

**Comment by jayeshkrishna** [comment]
A call shown below would result in passing a NULL start/count pointer,

```
ret = PIO_put_var(pio_file, pio_var, pval);
```

**Comment by jedwards4b** [comment]
good point, thanks

**Comment by edhartnett** [comment]
OK, the problem here is that this solution breaks reading scalar vars. To read scalar vars, we need to pass the NULLs to nc_get_vars().

I think the correct answer is to move the generation of the start/count arrays in a new function PIOc_get_var_tc(). Similar to PIOc_get_var1_tc(), it will do all the argument mangling that needs to be done before calling PIOc_get_vars_tc().

PIOc_get_vars_tc() will just pass along any NULLs. That way, PIOc_get_var() will work, but PIOc_get_vars_tc() will also work for scalar vars.

I am working on this now and will have a PR up in a while.

**Comment by jayeshkrishna** [comment]
@edhartnett : Can you create a new issue and refer to this issue (rather than reopening an old one)?

**Comment by edhartnett** [comment]
OK!
